benchmark/benchmarkTests.go

```diff
@@ -0,0 +1,338 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+        "fmt"
+	"testing"
+
+	"github.com/awslabs/soci-snapshotter/benchmark/framework"
+	"github.com/containerd/containerd"
+)
+
+func BenchmarkPullImageFromECR(b *testing.B, imageRef string) {
+	containerdProcess, err := getContainerdProcess() 
+	if err != nil {
+                fmt.Printf("Error Starting Containerd: %v\n", err)
+		b.Fatal(err)
+	}
+	defer containerdProcess.StopProcess()
+	b.ResetTimer()
+	_, err = containerdProcess.PullImageFromECR(imageRef, platform, awsSecretFile)
+	if err != nil {
+                fmt.Printf("Error Pulling Image: %v\n", err)
+		b.Fatal(err)
+	}
+	b.StopTimer()
+}
+
+func BenchmarkSociRPullPullImage(
+	b *testing.B,
+	imageRef string,
+	indexDigest string) {
+	containerdProcess, err := getContainerdProcess() 
+	if err != nil {
+                fmt.Printf("Failed to create containerd proc: %v\n", err)
+		b.Fatal(err)
+	}
+	defer containerdProcess.StopProcess()
+	sociProcess, err := getSociProcess()	
+        if err != nil {
+                fmt.Printf("Failed to create soci proc: %v\n", err)
+		b.Fatal(err)
+	}
+	defer sociProcess.StopProcess()
+	sociContainerdProc := SociContainerdProcess{containerdProcess}
+	b.ResetTimer()
```

Comment:
function contents above here are duplicated many places. defer() prevents straightforward refactor into a reusable function. is there another way to avoid this duplication?

Reply:
- You could tie the cancellation of the context to destruction of the resources.
- Or https://pkg.go.dev/testing#B.Cleanup